### ISSUE TITLE
Select Description Text

### ISSUE THREAD

**Issue by scurker** [dequelabs/cauldron#1656]
## Overview

`Select` should accept a new optional `description` property that is displayed immediately below the label text. When `description` is present, the associated input field should have `aria-describedby` with an idref pointing to the associated description.

## Related Components

- https://github.com/dequelabs/cauldron/issues/1641
- https://github.com/dequelabs/cauldron/issues/1658
- https://github.com/dequelabs/cauldron/issues/1657

### Visual Design

Figma: https://www.figma.com/design/CEFVdiecqDjLSjhorjHUzI/Product-Foundations-(Cauldron)-Library?node-id=50-1992&t=UIec9KMPGKQia7fz-4

![](https://github.com/user-attachments/assets/0b4458a4-ac61-4afb-a485-36e11731c571)

When the "label text" overflows with the "required" label, the label should wrap onto a new line without interfering with the placement or sizing of required.

> [!NOTE]
> This should match the description text that is current present on `RadioGroup`

### Interface / Props

```tsx
interface SelectProps {
  // ...other props
  description?: React.ReactNode
}
```

### Accessibility

> [!NOTE]
> When both `description` and `error` are set, the `aria-describedby` property should reference the description idref first, followed by the error idref.

## Additional Changes

With this change, the placement of the error message should also be visually located immediately above the select field with the addition of a `caution` icon preceding the error text.

The _Required_ text should not have any font styles applied and have italics removed.

**Comment by Amandeque** [comment]
Team Drums Estimation: 3 Drum Points

**Comment by Keerthi-Penukonda** [comment]
@DanylAksonov , could you please provide test notes for this ticket.
